src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -394,7 +482,21 @@ internal static class EventLogEntryCollectionExtensions
     {
         internal static EventLogEntry LastOrDefault(this EventLogEntryCollection elec)
         {
-            return elec.Count > 0 ? elec[elec.Count - 1] : null;
+            bool entryRetrieved = true;
+            EventLogEntry eventLogEntry = null;
+            while (entryRetrieved)
+            {
+                try
+                {
+                    eventLogEntry = elec.Count > 0 ? elec[elec.Count - 1] : null;
+                    entryRetrieved = false;
+                }
+                catch (Win32Exception)
```

Comment:
Suggestion: limit the catch to just Windows 7 (Use PlatformDetection.IsWindows7) since we have only seen it there. That will alert us if something new happens.